src/transport_performance/utils/defence.py

```diff
@@ -31,7 +31,7 @@ def _is_path_like(pth, param_nm):
 def _check_parent_dir_exists(pth, param_nm, create=False):
     _is_path_like(pth, param_nm)
     # replace back slashes to ensure consistency
-    pth = pth.replace("\\", "/").replace(repr("\\"), "").replace("'", "")
+    pth = str(pth).replace("\\", "/").replace(repr("\\"), "").replace("'", "")
```

Comment:
This isn't ideal but it's the only way to ensure paths are consistent and also that paths including singular \ (escape char) are correctly processed. It also provides the leniency for the users path to include n number of '\' in a path.
An example of this includes out_pth="D:\\\dir1\dir2\\dir3\dir4\file.html"